workflows/prognostic_c48_run/runtime/radiation_wrapper.py

```diff
@@ -0,0 +1,484 @@
+            )
+        out[name] = da.unstack(dim=sample_dim)
+    return out
+
+
+def _grid(state: State) -> Tuple[Mapping[str, np.ndarray], xr.DataArray]:
+    grid_names = ["longitude", "latitude"]
+    stacked_grid = _stack(
+        xr.Dataset({name: state[name] for name in grid_names}), sample_dim="column"
+    )
+    coords = stacked_grid.column
+    lon_rad = stacked_grid["longitude"].values
+    lat_rad = stacked_grid["latitude"].values
+    sinlat = np.sin(lat_rad)
+    coslat = np.cos(lat_rad)
+    return (
+        {"xlon": lon_rad, "xlat": lat_rad, "sinlat": sinlat, "coslat": coslat},
+        coords,
+    )
+
+
+SFC_VAR_MAPPING = {
+    "tsfc": "surface_temperature",
+    "slmsk": "land_sea_mask",
+    "snowd": "snow_depth_water_equivalent",
+    "sncovr": "snow_cover_in_fraction",
+    "snoalb": "maximum_snow_albedo_in_fraction",
+    "zorl": "surface_roughness",
+    "hprime": "orographic_variables",
+    "alvsf": "mean_visible_albedo_with_strong_cosz_dependency",
+    "alnsf": "mean_near_infrared_albedo_with_strong_cosz_dependency",
+    "alvwf": "mean_visible_albedo_with_weak_cosz_dependency",
+    "alnwf": "mean_near_infrared_albedo_with_weak_cosz_dependency",
+    "facsf": "fractional_coverage_with_strong_cosz_dependency",
+    "facwf": "fractional_coverage_with_weak_cosz_dependency",
+    "fice": "ice_fraction_over_open_water",
+    "tisfc": "surface_temperature_over_ice_fraction",
+}
+
+
+def _sfcprop(
+    state: State, sfc_var_mapping: Mapping[str, str] = SFC_VAR_MAPPING
+) -> Mapping[str, np.ndarray]:
+    sfc = xr.Dataset({k: state[v] for k, v in list(sfc_var_mapping.items())})
+    # we only want the first of "hprime" variables
+    sfc["hprime"] = sfc["hprime"].isel({"orographic_variable": 0})
+    stacked_sfc = _stack(sfc)
+    return {name: stacked_sfc[name].values for name in stacked_sfc.data_vars}
+
+
+OUT_NAMES = [
+    "clear_sky_downward_longwave_flux_at_surface_python",
+    "clear_sky_downward_shortwave_flux_at_surface_python",
+    "clear_sky_upward_longwave_flux_at_surface_python",
+    "clear_sky_upward_shortwave_flux_at_surface_python",
+    "clear_sky_upward_longwave_flux_at_top_of_atmosphere_python",
+    "clear_sky_upward_shortwave_flux_at_top_of_atmosphere_python",
+    "total_sky_downward_longwave_flux_at_surface_python",
+    "total_sky_downward_shortwave_flux_at_surface_python",
+    "total_sky_upward_longwave_flux_at_surface_python",
+    "total_sky_upward_shortwave_flux_at_surface_python",
+    "total_sky_downward_shortwave_flux_at_top_of_atmosphere_python",
+    "total_sky_upward_longwave_flux_at_top_of_atmosphere_python",
+    "total_sky_upward_shortwave_flux_at_top_of_atmosphere_python",
+]
+
+
+def _rename_out(
+    out: Tuple[Mapping[str, np.ndarray], ...], out_names: Sequence[str] = OUT_NAMES
+) -> Mapping[str, np.ndarray]:
+    radtendout, diagout, _ = out
+    renamed = {}
+    for name in out_names:
+        obj = radtendout if "surface" in name else diagout
```

Comment:
This parsing logic seems a bit brittle...maybe just write an explicit list/mapping someplace. Bonus points if it is something that can be static typed.